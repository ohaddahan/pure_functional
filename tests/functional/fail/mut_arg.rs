use functional_macro::pure_functional;

#[pure_functional]
fn test(i1: &mut String) {
    *i1 = "world".to_string();
}

fn main() {
    let mut i1 = "hello".to_string();
    test(&mut i1);
}
