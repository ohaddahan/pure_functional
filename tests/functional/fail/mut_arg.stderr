error: pure_functional: function with mutable arguments is not supported
 --> tests/functional/fail/mut_arg.rs:3:1
  |
3 | #[pure_functional]
  | ^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `pure_functional` (in Nightly builds, run with -Z macro-backtrace for more info)
